### PR TITLE
[eclipse/xtext#1679] port xtend code to java

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1679] port xtend code to java
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/GeneratorConfig.java [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+	 * <ul>
+	 * <li><code>${sourcefile}</code></li>
+	 * </ul>
+	 */
+	private String generatedAnnotationComment;
+
+	/**
+	 * Version of the Java Source code that's being generated by the compiler.
+	 */
+	private JavaVersion javaSourceVersion = JavaVersion.JAVA5;
```
Comment (tivervac) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/GeneratorConfig.java:52]: Is this still true though?

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/controlflow/ConstantConditionsInterpreter.xtend [hunk 2]
```diff
@@ -25,6 +25,7 @@ import org.eclipse.xtext.common.types.JvmFormalParameter
 import org.eclipse.xtext.common.types.JvmIdentifiableElement
 import org.eclipse.xtext.common.types.JvmMember
 import org.eclipse.xtext.common.types.JvmType
+import org.eclipse.xtext.resource.persistence.StorageAwareResource
```
Comment (tivervac) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/controlflow/ConstantConditionsInterpreter.xtend:28]: You can change the copyright date on this file though

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/arguments/IFeatureCallArgumentSlot.java [hunk 3]
```diff
@@ -17,27 +17,26 @@
...
+	boolean isSuperfluous();
+
+	/**
+	 * The list of expressions. The list is never <code>null</code> but may
+	 * contain <code>null</code> entries.
+	 */
+	List<XExpression> getArgumentExpressions();
+
+	/**
+	 * @Nullable
```
Comment (tivervac) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/arguments/IFeatureCallArgumentSlot.java:32]: Don't we have the actual `@Nullable` available?

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/arguments/IFeatureCallArgumentSlot.java [hunk 4]
```diff
@@ -17,27 +17,26 @@
  * @noextend This interface is not intended to be extended by clients.
  * @noimplement This interface is not intended to be implemented by clients.
  */
-@SuppressWarnings("all")
 public interface IFeatureCallArgumentSlot {
```
Comment (tivervac) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/arguments/IFeatureCallArgumentSlot.java:20]: You can update the copyright date on here as well

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/util/TypesOrderUtil.java [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+import java.util.Collection;
+
+import org.eclipse.xtext.xbase.typesystem.conformance.TypeConformanceComputer;
+import org.eclipse.xtext.xbase.typesystem.references.LightweightTypeReference;
+import org.eclipse.xtext.xbase.typesystem.util.CommonTypeComputationServices;
+
+import com.google.inject.Inject;
+
+/**
+ * Small utilitiy class that allows to check whether a given type was already handled in a chain
```
Comment (tivervac) [inline org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/util/TypesOrderUtil.java:22]: s/utilitiy/utility/

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/GeneratorConfig.java (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/GeneratorConfig.java:52]: still need help with https://github.com/eclipse/xtext-extras/issues/113

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/arguments/IFeatureCallArgumentSlot.java (hunk 3 above)
Comment (cdietrich) [reply org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/arguments/IFeatureCallArgumentSlot.java:32]: unfortunately not. at least not ubiquitous

**Comment by cdietrich** [comment]
@tivervac pr is updated

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/arguments/IFeatureCallArgumentSlot.java (hunk 3 above)
Comment (tivervac) [reply org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/typesystem/arguments/IFeatureCallArgumentSlot.java:32]: Sad life

File: org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/GeneratorConfig.java (hunk 1 above)
Comment (tivervac) [reply org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/GeneratorConfig.java:52]: Okidoki

**Pull request merged**